python/lsst/cp/pipe/makeBrighterFatterKernel.py

```diff
@@ -0,0 +1,1380 @@
+        return pipeBase.TaskRunner.getTargetList(parsedCmd, visitPairs=visitPairs, **kwargs)
+
+
+class BrighterFatterKernelTaskDataIdContainer(pipeBase.DataIdContainer):
+    """A DataIdContainer for the MakeBrighterFatterKernelTask."""
+
+    def makeDataRefList(self, namespace):
+        """Compute refList based on idList.
+
+        This method must be defined as the dataset does not exist before this
+        task is run.
+
+        Parameters
+        ----------
+        namespace
+            Results of parsing command-line
+            (with ``butler`` and ``log`` elements).
+
+        Notes
+        -----
+        Not called if ``add_id_argument`` called
+        with ``doMakeDataRefList=False``.
+        Note that this is almost a copy-and-paste of the vanilla implementation,
+        but without checking if the datasets already exist,
+        as this task exists to make them.
+        """
+        if self.datasetType is None:
+            raise RuntimeError("Must call setDatasetType first")
+        butler = namespace.butler
+        for dataId in self.idList:
+            refList = list(butler.subset(datasetType=self.datasetType, level=self.level, dataId=dataId))
+            # exclude nonexistent data
+            # this is a recursive test, e.g. for the sake of "raw" data
+            if not refList:
+                namespace.log.warn("No data found for dataId=%s", dataId)
+                continue
+            self.refList += refList
+
+
+class MakeBrighterFatterKernelTask(pipeBase.CmdLineTask):
+    """Brighter-fatter effect correction-kernel calculation task.
+
+    A command line task for calculating the brighter-fatter correction
+    kernel from pairs of flat-field images (with the same exposure length).
+
+    The following operations are performed:
+
+    - The configurable isr task is called, which unpersists and assembles the
+      raw images, and performs the selected instrument signature removal tasks.
+      For the purpose of brighter-fatter coefficient calculation is it
+      essential that certain components of isr are *not* performed, and
+      recommended that certain others are. The task checks the selected isr
+      configuration before it is run, and if forbidden components have been
+      selected task will raise, and if recommended ones have not been selected,
+      warnings are logged.
+
+    - The gain of the each amplifier in the detector is calculated using
+      the photon transfer curve (PTC) method and used to correct the images
+      so that all calculations are done in units of electrons, and so that the
+      level across amplifier boundaries is continuous.
+      Outliers in the PTC are iteratively rejected
+      before fitting, with the nSigma rejection level set by
+      config.nSigmaClipRegression. Individual pixels are masked from
+      the image based on config.nSigmaClipGainCalc
```

Comment:
I'm afraid I don't understand this last sentence. Is it practical to be a bit more specific about what calculation is used to mask these pixels? Also please document the following here: is the mask of the exposure updated by this operation? If so, which mask plane is updated (and if it's one that you add, say that as well)?

Reply:
Sorry, I see how this was confusing - this is actually a comment on the input images, and I've changed it to say that. Also, nothing is masked or updated, and I've also changed the text to not use the work "mask".